discovery_child_development/notebooks/04_train_initial_model.py

```diff
@@ -0,0 +1,444 @@
+# %% [markdown]
+# # Train a baseline model and log on wandb
+#
+# In this notebook we:
+# * Perform some EDA on the (training) data - check distribution of scores and then use this to filter the data; check distributions of OpenAlex concepts and the sub-categories that we assigned
+# * Turn the data into the correct format for training classification models (ie a one-hot-encoding representation of the target)
+# * Define a baseline classifier than guesses that all new publications will be tagged with the most common combination of labels that we have in the data
+# * Log the model and key metrics on wandb
+
+# %%
+# Load libraries
+import pandas as pd
+import numpy as np
+import os
+import pickle
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import (
+    accuracy_score,
+    precision_score,
+    recall_score,
+    f1_score,
+    hamming_loss,
+    jaccard_score,
+    classification_report,
+)
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+## plotting libraries
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import wandb
+
+## nesta ds
+from nesta_ds_utils.loading_saving import S3
+
+## project code
+from discovery_child_development import PROJECT_DIR, logging
+from discovery_child_development.utils import bigquery, labelling_utils
+from discovery_child_development.utils import wandb as wb
+from discovery_child_development.utils.io import import_config
+
+bigquery.find_credentials("GOOGLE_SHEETS_CREDENTIALS")
+
+## constants
+
+S3_BUCKET = os.environ["S3_BUCKET"]
+PARAMS = import_config("config.yaml")
+CONCEPT_IDS = "|".join(PARAMS["openalex_concepts"])
+INPUT_PATH = "data/openAlex/processed/"
+DATA_PATH_LOCAL = os.path.join(PROJECT_DIR, "inputs", "data")
+FIG_PATH = os.path.join(PROJECT_DIR, "outputs", "figures")
+MODEL_PATH = os.path.join(PROJECT_DIR, "outputs", "models")
+SEED = 42
+
+## variables
+SCORE_THRESHOLD = 0.3  # we will remove any concepts (and corresponding subcategories) assigned with less than 0.3 confidence by the OpenAlex algorithm
+
+
+# %%
+# helpful functions
+class MostCommonClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self, labels):
+        self.labels = labels
+
+    def fit(self, X, y=None):
+        # Nothing happens here because it doesn't learn from the data
+        return self
+
+    def predict(self, X):
+        # Returns the most common label combination for all input
+        return [self.labels for _ in range(len(X))]
```

Comment:
Not sure what happens here - you have a list of elements where each element is self.labels?

Reply:
Oh I think I got it - but would be good if you have a docstring somewhere explaining what this is